### DISCUSSION TITLE
Expansion of models for time-series such as transformers or denoising

### DISCUSSION THREAD

**Discussion by Infinitay** [pycaret/pycaret#3243] [Ideas]
First and foremost, I'd like to apologize if these models are already offered. I'm still learning so I'm not well-versed in the field of ML. [I looked over the API and I couldn't spot any estimator](https://pycaret.readthedocs.io/en/latest/api/time_series.html#pycaret.time_series.create_model) that seemed to reference mention of the mentioned models in the table below. 

I was wondering if PyCaret could be expanded to support some of the latest transformer or denoising models for time series forecasting. I came across a few recent models that seems to perform really well on their benchmarks compared to other methods. If I recall correctly, PyCaret currently only supports univariate, so I will provide the following benchmark:

![image](https://user-images.githubusercontent.com/6964154/211793126-3de597c0-a727-487f-8f32-450e415926d8.png)
_source: [Towards Long-Term Time-Series Forecasting: Feature, Pattern, and Distribution](https://arxiv.org/pdf/2301.02068v1.pdf)_

The latest Conformer model seems to outperform most models. It would be nice to be able to use this as an estimator within PyCaret. Even-so, both the Autoformer and Informer models still perform well. Further more, [according to another benchmark on the same ETT datasets](https://github.com/cure-lab/LTSF-Linear/blob/main/LTSF-Benchmark.md#univariate-long-term-series-forecasting-results-on-four-ett-datasets-with-input-length-i--96-and-prediction-length-o-96-192-336-720), it seems like Linear models such as NLinear and DLinear outperform Autoformer and Informer, and further shown on [PapersWithCode's benchmark](https://paperswithcode.com/sota/time-series-forecasting-on-etth1-720).

In addition to a new transformer model, Conformer, Paddle has also worked on using denoising for time-series forecasting as they describe in their [paper](https://arxiv.org/pdf/2301.03028v1.pdf). Although, it doesn't seem like it outperforms their Conformer model.

**tl;dr would it be possible for PyCaret to expand the current time-series estimators to include some of the _new_ transformer models such as Conformer and possibly the denoising model D3VAE?**

| Model | Method | Paper | Code Repository | Notes
|---|---|---|---|---|
| **Conformer** | Transformer | [Towards Long-Term Time-Series Forecasting: Feature, Pattern, and Distribution](https://arxiv.org/pdf/2301.02068v1.pdf) | https://github.com/PaddlePaddle/PaddleSpatial/tree/main/research/Conformer | Code is not available as of 11/1/2022
| Autoformer | Transformer | [Autoformer: Decomposition Transformers with Auto-Correlation for Long-Term Series Forecasting](https://arxiv.org/pdf/2106.13008v5.pdf) | https://github.com/thuml/autoformer | Also available at https://github.com/cure-lab/LTSF-Linear |
| Informer | Transformer | [Informer: Beyond Efficient Transformer for Long Sequence Time-Series Forecasting](https://arxiv.org/pdf/2012.07436v3.pdf) | https://github.com/zhouhaoyi/Informer2020 | Also available at https://github.com/cure-lab/LTSF-Linear |
| **NLinear** | Linear? | [Are Transformers Effective for Time Series Forecasting?](https://arxiv.org/pdf/2205.13504v3.pdf) | https://github.com/cure-lab/LTSF-Linear | NLinear seems to be better than DLinear for univariate |
| **DLinear** | Linear? | [Are Transformers Effective for Time Series Forecasting?](https://arxiv.org/pdf/2205.13504v3.pdf) | https://github.com/cure-lab/LTSF-Linear | Good for multivariate |
| **D3VAE** | Denoising | [Generative Time Series Forecasting with Diffusion, Denoise, and Disentanglement](https://arxiv.org/pdf/2301.03028v1.pdf) | https://github.com/PaddlePaddle/PaddleSpatial/tree/main/research/D3VAE | Depends on PaddlePaddle |
